pkg/pushsync/pushsync.go

```diff
@@ -147,10 +166,98 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 				return fmt.Errorf("chunk store: %w", err)
 			}
 
+			count := 0
+			// Push the chunk to some peers in the neighborhood in parallel for replication.
+			// Any errors here should NOT impact the rest of the handler.
+			err = ps.topologyDriver.EachNeighbor(func(peer swarm.Address, po uint8) (bool, bool, error) {
+
+				// skip forwarding peer
+				if peer.Equal(p.Address) {
+					return false, false, nil
+				}
+
+				if count == nPeersToPushsync {
+					return true, false, nil
+				}
+				count++
+
+				go func(peer swarm.Address) {
+
+					var err error
+					defer func() {
+						if err != nil {
+							ps.logger.Tracef("pushsync replication: %v", err)
+						}
+					}()
+
+					// price for neighborhood replication
+					const receiptPrice uint64 = 0
+
+					headers, err := headerutils.MakePricingHeaders(receiptPrice, chunk.Address())
+					if err != nil {
+						err = fmt.Errorf("make pricing headers: %w", err)
+						return
+					}
+
+					streamer, err := ps.streamer.NewStream(ctx, peer, headers, protocolName, protocolVersion, streamName)
+					if err != nil {
+						err = fmt.Errorf("new stream for peer %s: %w", peer.String(), err)
+						return
+					}
+					defer streamer.Close()
+
+					returnedHeaders := streamer.Headers()
+					_, returnedPrice, _, err := headerutils.ParsePricingResponseHeaders(returnedHeaders)
+					if err != nil {
+						err = fmt.Errorf("push price headers read returned: %w", err)
+						return
+					}
+
+					// check if returned price matches presumed price, if not, return early
+					if returnedPrice != receiptPrice {
```

Comment:
if the returned price does not match the previously calculated price, we want some things to happen, but not to return early based on this condition.

- we want to notifypeerprice with the index included, as in other places.
- we want to set receiptprice to returnedprice (as in other occurences)

after this, because we are pushsyncing in the neighborhood, and nobody will cover our cost of this (as it is supposed to be free) if the receiptPrice after the header communication is larger than zero, we don't want to push to that peer (instead we want to return in this case and reset the stream).

So because of this, we don't need the reserve, release, credit, because if there was  non zero value to reserve/release/credit, we would return early